bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/services/DynamicDeckService.java

```diff
@@ -0,0 +1,48 @@
+package com.adobe.acs.commons.indesign.dynamicdeckdynamo.services;
+
+import com.adobe.acs.commons.indesign.dynamicdeckdynamo.exception.DynamicDeckDynamoException;
+import org.apache.sling.api.resource.Resource;
+import org.apache.sling.api.resource.ResourceResolver;
+
+import javax.jcr.RepositoryException;
+import java.io.IOException;
+import java.util.List;
+
+public interface DynamicDeckService {
+
+
+    /**
+     * This method will create the InDesign deck based on list of assets and other supporting parameter supplied.
+     * <p>
+     * TODO Planning to create service account and remove the dependency of resourceResolver from this service
+     *
+     * @param deckName
+     * @param masterAssetPath
+     * @param assetResourceList
+     * @param templateFolderPath
+     * @param resourceResolver
+     * @param destinationFolderPath
+     * @return
+     * @throws IOException
+     * @throws RepositoryException
+     */
+    public String createDeck(String deckName, String masterAssetPath, List<Resource> assetResourceList,
+                             String templateFolderPath, String destinationFolderPath, ResourceResolver resourceResolver)
```

Comment:
Why do you provide paths as strings? Why not Resources? Then you don't need to care about a resourceresolver as well (it's implicit then).